### ISSUE TITLE
paddle.static.nn.sequence_unpad在某些情况下core

### ISSUE THREAD

**Issue by yzhw9981** [PaddlePaddle/Paddle#45725]
### bug描述 Describe the Bug

```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
########################################################################
# 
# Copyright (c) 2022 Baidu.com, Inc. All Rights Reserved
# 
########################################################################
 
"""
File: test.py
Author: work(work@baidu.com)
Date: 2022/06/24 19:35:22
Brief: 
"""
import paddle
from paddle import fluid
paddle.enable_static()

if __name__ == "__main__":
    exe = paddle.static.Executor(paddle.CPUPlace())
    startup_program = paddle.static.default_startup_program()
    time_slice = 2
    seq = paddle.randn([2, 4, 16])
    combine_output = paddle.assign([2, 2])

    #下面两行代码会core，交换位置不core
    seq = paddle.static.nn.fc(x=seq, size=16, num_flatten_dims=2)
    emb = paddle.static.nn.sequence_unpad(seq, combine_output)

    out, _ = paddle.fluid.layers.dynamic_lstm(emb, size=16)
    out = paddle.static.nn.sequence_last_step(out)
    out = paddle.static.nn.fc(x=out, size=2)
    loss = paddle.nn.functional.softmax_with_cross_entropy(out, paddle.ones([2, 1], dtype='int64'))
    loss = paddle.mean(loss)
    optimizer = paddle.optimizer.Adagrad(learning_rate=0.001)
    optimizer.minimize(loss)
    exe.run(startup_program)
    exe.run() 
# vim: set ts=4 sw=4 sts=4 tw=100: 
```

### 其他补充信息 Additional Supplementary Information

paddle 2.3.0
CPU&GPU环境均复现

**Comment by wangna11BD** [comment]
您好～首先我们不推荐使用 sequence 系列的 API 了。如果您需要用的是 LSTM 可以试试使用 paddle.nn.LSTM.
我将代码改写了一下，仅测试 sequence_unpad, 并无出现问题，可能是后续代码的问题。
import numpy as np
import paddle
from paddle import fluid
paddle.enable_static()

if __name__ == "__main__":
exe = paddle.static.Executor(paddle.CPUPlace())
startup_program = paddle.static.default_startup_program()
time_slice = 2
seq = paddle.randn([2, 4, 16])
combine_output = paddle.assign([2, 2])

#下面两行代码会core，交换位置不core
seq = paddle.static.nn.fc(x=seq, size=16, num_flatten_dims=2)
emb = paddle.static.nn.sequence_unpad(seq, combine_output)

#out, _ = paddle.fluid.layers.dynamic_lstm(emb, size=16)
#out = paddle.static.nn.sequence_last_step(out)
#out = paddle.static.nn.fc(x=out, size=2)
#loss = paddle.nn.functional.softmax_with_cross_entropy(out, paddle.ones([2, 1], dtype='int64'))
loss = paddle.mean(emb)
optimizer = paddle.optimizer.Adagrad(learning_rate=0.001)
optimizer.minimize(loss)
exe.run(startup_program)
emb_tensor, = exe.run(fetch_list=[emb], return_numpy=False)
print(np.array(emb_tensor))
